commons/zenoh-shm/src/posix_shm/segment.rs

```diff
@@ -12,150 +12,86 @@
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
 
-use std::fmt::{Debug, Display};
+use std::{fmt::Debug, num::NonZeroUsize};
 
 use rand::Rng;
-use shared_memory::{Shmem, ShmemConf, ShmemError};
-use zenoh_result::{bail, zerror, ZResult};
+use zenoh_result::{bail, ZResult};
 
-#[cfg(target_os = "linux")]
-use super::segment_lock::unix::{ExclusiveShmLock, ShmLock};
-use crate::cleanup::CLEANUP;
+use crate::{cleanup::CLEANUP, shm};
 
 const SEGMENT_DEDICATE_TRIES: usize = 100;
-const ECMA: crc::Crc<u64> = crc::Crc::<u64>::new(&crc::CRC_64_ECMA_182);
 
 /// Segment of shared memory identified by an ID
 pub struct Segment<ID>
 where
     rand::distributions::Standard: rand::distributions::Distribution<ID>,
-    ID: Clone + Display,
+    ID: shm::SegmentID,
 {
-    shmem: Shmem, // <-------------|
-    id: ID,       //               |
-    #[cfg(target_os = "linux")] // | location of these two fields matters!
-    _lock: Option<ShmLock>, // <---|
-}
-
-#[cfg(target_os = "linux")]
-impl<ID> Drop for Segment<ID>
-where
-    rand::distributions::Standard: rand::distributions::Distribution<ID>,
-    ID: Clone + Display,
-{
-    fn drop(&mut self) {
-        if let Some(lock) = self._lock.take() {
-            if let Ok(_exclusive) = std::convert::TryInto::<ExclusiveShmLock>::try_into(lock) {
-                // in case if we are the last holder of this segment make ourselves owner to unlink it
-                self.shmem.set_owner(true);
-            }
-        }
-    }
+    shmem: shm::Segment<ID>,
```

Comment:
1. It's confusing that we have `Segment` and `shm::Segment` simultaneously.
2. To me, `shm::Segment` is redundant and can be simplified to the inner `platform::SegmentImpl<ID>`. What do you think?